src/App.tsx

```diff
@@ -2,24 +1,49 @@
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [selectedPerPageValue, setSelectedPerPageValue] = useState<number>(5);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const handlePerPageChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedPerPageValue(Number(event.target.value));
+    setCurrentPage(1);
+  };
+
+  const onPageChange = (page: number) => {
+    setCurrentPage(page);
+  };
+
+  const indexOfLastItem = currentPage * selectedPerPageValue;
+  const indexOfFirstItem = indexOfLastItem - selectedPerPageValue;
+
+  const currentItems = items.slice(indexOfFirstItem, indexOfLastItem);
+
+  const itemFrom = indexOfFirstItem + 1;
+  const itemTo = Math.min(indexOfLastItem, items.length);
```

Comment:
The calculation of `indexOfLastItem`, `indexOfFirstItem`, `itemFrom`, and `itemTo` can be part of the pagination logic. If you move these calculations into a custom hook or a utility function, the `App` component will be more readable and the logic can be easily reused in other components if needed.